### DISCUSSION TITLE
Mainboard upgrade recommendations for Ender 3 S1 to use the features that require more space like Linear Advance, Input Shaping, UBL

### DISCUSSION THREAD

**Discussion by Luro02** [mriscoc/Ender3V2S1#1046] [Q&A]
My Ender 3 S1 (sadly) has an SMT32F401 RCT6, which has limited space for the firmware. This prevents me from using some Marlin features like Linear Advance **and** UBL.

The new release mentions
> Preliminary support for E3 Free-runs board CR4NTxxC10 and other boards https://store.creality.com/products/e3-free-runs-tmc2209-32-bit-open-source-silent-motherboard

so this made me think that I could upgrade my mainboard to have more space for the firmware.

Obviously not every mainboard is compatible with this firmware, which is why I am asking if there
are there any recommendations for boards that work with the Ender 3 S1 and have enough space to comfortably fit future marlin features?
Does the board in the link work? (I guess not, because it has an SMT32F401 as well?)
What do you have to look for?

**Comment by mriscoc** [comment]
Hi, the E3 Free-runs board has a F4 but RET6 so, there is more program memory and ram space. Sadly, I don't have that board to do the proper tests.
